storage/mysql/log_storage.go

```diff
@@ -320,11 +325,17 @@ func (t *logTreeTX) QueueLeaves(leaves []*trillian.LogLeaf, queueTimestamp time.
 		insertSQL = insertUnsequencedLeafSQLNoDuplicates
 	}
 
-	for i, leaf := range leaves {
+	// Insert in order of the hash values in the leaves.
+	orderedLeaves := make([]*trillian.LogLeaf, len(leaves))
+	copy(orderedLeaves, leaves)
```

Comment:
Not copying will modify the input parameters, which I prefer to avoid (and we're just copying pointers not objects so it should be too expensive).

I guess we could add comments to indicate that this entrypoint may modify the passed-in slice if you're super keen to avoid the copy.